docker/update_or_build.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+function usage {
+  echo "Usage:"
+  echo "build and properly tags images for deployment to docker hub"
+  echo "update_or_build.sh [-p] -g <GITHASH> -b <TARGET_BRANCH> -n <image name> [-u]"
+  echo "-p indicates this a prebuild, where images are built and pushed to dockerhub with an infix of '_pre_', should be run on the 'auto' branch, trigger by bors."
+  echo "-b the branch we're building on, or the branch we're targeting if a prebuild"
+  echo "-n name, one of init, mint, validator, validator-dynamic, safety-rules, cluster-test"
+  echo "-u 'upload', or 'push' the docker images will be pushed to dockerhub, otherwise only locally tag"
+  echo "should be called from the root folder of the libra project, and must have it's .git history"
+}
+
+PREBUILD=false;
+NAME=
+BRANCH=
+PUSH=false
+
+#parse args
+while getopts "pb:n:u" arg; do
+  case $arg in
+    p)
+      PREBUILD="true"
+      ;;
+    n)
+      NAME=$OPTARG
+      ;;
+    b)
+      BRANCH=$OPTARG
+      ;;
+    u)
+      PUSH="true"
+      ;;
+    h)
+      usage;
+      exit 0;
+      ;;
+  esac
+done
+
+GIT_REV=$(git rev-parse --short=8 HEAD)
+
+[ "$BRANCH" != "" ] || { echo "-b branch must be set"; usage; exit 99; }
+[ "$GIT_REV" != "" ] || { echo "Could not determine git revision, aborting"; usage; exit 99; }
+[ "$NAME" != "" ] || { echo "-n name must be set"; usage; exit 99; }
+
+PULLED="-1"
+
+#Convert dashes to underscores to get tag names, except for validator, which for some reason is "e2e"
+tag_name=`echo $NAME | sed 's/-/_/g'`
+if [ $NAME == "validator" ]; then
+  tag_name="e2e";
+fi
+
+#The name of the docker image built in the "auto" branch
+PRE_NAME=libra/test:libra_${tag_name}_pre_${BRANCH}_${GIT_REV}
+
+#If not a prebuild *attempt* to pull the previously built image.
+if [ $PREBUILD != "true" ]; then
+  docker pull ${PRE_NAME}
+  export PULLED=$?
+fi
+
+#if a prebuild, always -1, else if "docker pull" failed build the image.
+if [ "$PULLED" != "0" ]; then
+  docker/$NAME/build.sh
+  #tag the built image with consistent naming...  ugh cluster-test, and e2e
```

Comment:
We can skip `cluster-test` and `libra_e2e`.

Reply:
want cluster test for the every hour node health check in premainnet

Reply:
Will defer to you since you had a plan to use this image.